src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,56 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final String HEADER_TO_START_MESSAGE = "Report for period ";
+    private static final String DATE_FORMAT = "dd.MM.yyyy";
+    private static final String WRONG_DATE_FORMAT_MESSAGE = "Incorrect date format. "
+            + "The date format should be " + DATE_FORMAT;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder salaryInfo = new StringBuilder(HEADER_TO_START_MESSAGE)
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+        for (String name : names) {
+            salaryInfo
+                    .append(System.lineSeparator())
+                    .append(name)
+                    .append(" - ")
+                    .append(getSalaryByName(name, data, dateFrom, dateTo));
+        }
+        return salaryInfo.toString();
+    }
+
+    private int getSalaryByName(String name, String[] data, String dateFrom, String dateTo) {
+        int sumSalary = 0;
+        for (String itemData : data) {
+            EmployeeWorkingTimeData employeeData =
+                    new EmployeeWorkingTimeData(itemData.split(" "));
+            if (name.equals(employeeData.getName())
+                    && isDateInRange(employeeData.getWorkingDay(), dateFrom, dateTo)) {
+                sumSalary += employeeData.getRatePerHour() * employeeData.getNumberOfHours();
+            }
+        }
+        return sumSalary;
+    }
+
+    private LocalDate getDateFromString(String stringDate) {
```

Comment:
methods should be ordered in order of their calls

Reply:
Ok, it is corrected.